### ISSUE TITLE
ome.zarr datatype

### ISSUE THREAD

**Issue by tischi** [tischi/i2k-2020-s3-zarr-workshop#9]
@joshmoore @constantinpape 
It looks as if the current specification of ome.zarr allows, in principle, for each resolution layer to have a different data type?
At least on the highest level the `dtype` is not specified. Is that on purpose?
If so, I would have to hack it and read the `dtype` from one of the resolution layers, just assuming that it is the same for the other resolutions.
Opinions?

**Comment by constantinpape** [comment]
> It looks as if the current specification of ome.zarr allows, in principle, for each resolution layer to have a different data type?

I don't think that should be allowed. If it's not defined in the spec yet, it should maybe be added that all resolution layers should have the same datatype.

> If so, I would have to hack it and read the `dtype` from one of the resolution layers, just assuming that it is the same for the other resolutions.

I think that approach is fine. You can also explicitly check in the setup that all scales have the same dtype and raise an error if they don't.

**Comment by joshmoore** [comment]
Thanks for the heads up. Something we missed. (Opening a repo issue to address this)
